diff --git a/node_modules/expo-file-system/ios/Legacy/FileSystemLegacyModule.swift b/node_modules/expo-file-system/ios/Legacy/FileSystemLegacyModule.swift
index 5604d61..6d3ee1a 100644
--- a/node_modules/expo-file-system/ios/Legacy/FileSystemLegacyModule.swift
+++ b/node_modules/expo-file-system/ios/Legacy/FileSystemLegacyModule.swift
@@ -45,9 +45,9 @@ public final class FileSystemLegacyModule: Module {
       let optionsDict = options.toDictionary(appContext: appContext)
       switch url.scheme {
       case "file":
-        EXFileSystemLocalFileHandler.getInfoForFile(url, withOptions: optionsDict, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+        EXFileSystemLocalFileHandler.getInfoForFile(url, withOptions: optionsDict,         resolver: promise.legacyResolver, rejecter: promise.legacyRejecter)
       case "assets-library", "ph":
-        EXFileSystemAssetLibraryHandler.getInfoForFile(url, withOptions: optionsDict, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+        EXFileSystemAssetLibraryHandler.getInfoForFile(url, withOptions: optionsDict,         resolver: promise.legacyResolver, rejecter: promise.legacyRejecter)
       default:
         throw UnsupportedSchemeException(url.scheme)
       }
@@ -117,9 +117,9 @@ public final class FileSystemLegacyModule: Module {
       try ensurePathPermission(appContext, path: toUrl.path, flag: .write)
 
       if fromUrl.scheme == "file" {
-        EXFileSystemLocalFileHandler.copy(from: fromUrl, to: toUrl, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+        EXFileSystemLocalFileHandler.copy(from: fromUrl, to: toUrl,         resolver: promise.legacyResolver, rejecter: promise.legacyRejecter)
       } else if ["ph", "assets-library"].contains(fromUrl.scheme) {
-        EXFileSystemAssetLibraryHandler.copy(from: fromUrl, to: toUrl, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+        EXFileSystemAssetLibraryHandler.copy(from: fromUrl, to: toUrl,         resolver: promise.legacyResolver, rejecter: promise.legacyRejecter)
       } else {
         throw InvalidFileUrlException(fromUrl)
       }
@@ -149,14 +149,14 @@ public final class FileSystemLegacyModule: Module {
 
       if sourceUrl.isFileURL {
         try ensurePathPermission(appContext, path: sourceUrl.path, flag: .read)
-        EXFileSystemLocalFileHandler.copy(from: sourceUrl, to: localUrl, resolver: promise.resolver, rejecter: promise.legacyRejecter)
+        EXFileSystemLocalFileHandler.copy(from: sourceUrl, to: localUrl,         resolver: promise.legacyResolver, rejecter: promise.legacyRejecter)
         return
       }
       let session = options.sessionType == .background ? backgroundSession : foregroundSession
       let request = createUrlRequest(url: sourceUrl, headers: options.headers)
       let downloadTask = session.downloadTask(with: request)
       let taskDelegate = EXSessionDownloadTaskDelegate(
-        resolve: promise.resolver,
+        resolve: promise.legacyResolver,
         reject: promise.legacyRejecter,
         localUrl: localUrl,
         shouldCalculateMd5: options.md5
@@ -175,7 +175,7 @@ public final class FileSystemLegacyModule: Module {
       }
       let session = options.sessionType == .background ? backgroundSession : foregroundSession
       let task = try createUploadTask(session: session, targetUrl: targetUrl, sourceUrl: localUrl, options: options)
-      let taskDelegate = EXSessionUploadTaskDelegate(resolve: promise.resolver, reject: promise.legacyRejecter)
+      let taskDelegate = EXSessionUploadTaskDelegate(resolve: promise.legacyResolver, reject: promise.legacyRejecter)
 
       sessionTaskDispatcher.register(taskDelegate, for: task)
       task.resume()
@@ -194,7 +194,7 @@ public final class FileSystemLegacyModule: Module {
         ])
       }
       let taskDelegate = EXSessionCancelableUploadTaskDelegate(
-        resolve: promise.resolver,
+        resolve: promise.legacyResolver,
         reject: promise.legacyRejecter,
         onSendCallback: onSend,
         resumableManager: taskHandlersManager,
@@ -231,7 +231,7 @@ public final class FileSystemLegacyModule: Module {
       }
 
       let taskDelegate = EXSessionResumableDownloadTaskDelegate(
-        resolve: promise.resolver,
+        resolve: promise.legacyResolver,
         reject: promise.legacyRejecter,
         localUrl: localUrl,
         shouldCalculateMd5: options.md5,
