diff --git a/ios/EXDefines.h b/ios/EXDefines.h
index 77e4570..c41654c 100644
--- a/ios/EXDefines.h
+++ b/ios/EXDefines.h
@@ -76,6 +76,35 @@
 // Converts nil -> [NSNull null]
 #define EXNullIfNil(value) (value ?: [NSNull null])
 
+static inline NSError *EXErrorWithMessage(NSString *message) {
+  return [NSError errorWithDomain:@"expo" code:-1 userInfo:@{NSLocalizedDescriptionKey: message}];
+}
+
+static inline void EXFatal(NSError *error) {
+  NSLog(@"EXFatal: %@", error.localizedDescription);
+}
+
+static inline void EXLogInfo(NSString *format, ...) {
+  va_list args;
+  va_start(args, format);
+  NSLogv(format, args);
+  va_end(args);
+}
+
+static inline void EXLogWarn(NSString *format, ...) {
+  va_list args;
+  va_start(args, format);
+  NSLogv(format, args);
+  va_end(args);
+}
+
+static inline void EXLogError(NSString *format, ...) {
+  va_list args;
+  va_start(args, format);
+  NSLogv(format, args);
+  va_end(args);
+}
+
 #import <ExpoModulesCore/Platform.h>
 
 #import <Foundation/Foundation.h>
